.github/workflows/main.yml

```diff
@@ -0,0 +1,16 @@
+name: test_sphinx_build
+on: [push, pull_request]
+
+jobs:
+  build_docs_job:
+    runs-on: ubuntu-22.04
+    name: build sphinx docs
+    steps:
+      - uses: actions/checkout@v3
+      - name: install sphinx
+        run: |
+          python3 -m pip install sphinx==4.3.0
```

Comment:
I would use the `requirements.txt` file in here as well, then we can have a single source for the build environment.